src/pages/MemberProfilePage.tsx

```diff
@@ -44,13 +54,15 @@ const Widgets: React.FC<{ member: Member }> = ({ member }) => {
       <Stack direction={{ base: 'column', md: 'row' }} spacing={4}>
         {member.leetcode && (
           <Box p={4} flex="1">
-            <LeetcodeProfile username={member.leetcode.username} />
+            <LeetcodeProfile
+              username={getUserFromUrlIfThisIsAUrl(member.leetcode.username)}
```

Comment:
Instead of changing text why not make the text box red if it's a url and tell the user to enter username or something.